Util/Xamarin.Build.Download/source/Xamarin.Build.Download/XamarinDownloadArchives.cs

```diff
@@ -287,10 +287,22 @@ ProcessStartInfo CreateExtractionArgs (string file, string contentDir, ArchiveKi
 			default:
 				throw new ArgumentException ("kind");
 			}
-			return new ProcessStartInfo (args.ProcessPath, args.ToString ()) {
-				WorkingDirectory = contentDir,
-				CreateNoWindow = true
-			};
+
+			ProcessStartInfo psi = null;
+			if (Platform.IsWindows)
+				psi = new ProcessStartInfo (args.ProcessPath, args.ToString ())
+				{
+					WorkingDirectory = null,
```

Comment:
could you provide link to docs for this trick, please?

Reply:
Setting to `null` will default to `System32` as the working directory, see [here](https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.processstartinfo.workingdirectory?view=net-5.0#remarks). This should be fine as no user/pass is specified & all the paths for the process and extraction are absolute. If we set it to `contentDir` it'll fail for paths over 260 chars.